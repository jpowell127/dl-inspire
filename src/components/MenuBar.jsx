import React from "react";
import PropTypes from "prop-types";
import Search from "./Search";
import { Menu, Icon, Header } from "semantic-ui-react";

function MenuBar({ handleSearchTermChange, getUsers }) {
  return (
    <Menu inverted size="small">
      <Menu.Item>
        <Header as="h1" color="pink">
          <Icon name="behance" />
          <Header.Content>Search</Header.Content>
        </Header>
      </Menu.Item>
      <Menu.Menu position="right">
        <Menu.Item>
          <Search
            handleSearchTermChange={handleSearchTermChange}
            getUsers={getUsers}
          />
        </Menu.Item>
      </Menu.Menu>
    </Menu>
  );
}

Search.MenuBar = {
  getUsers: PropTypes.func.isRequired,
  handleSearchTermChange: PropTypes.func.isRequired
};

export default MenuBar;
