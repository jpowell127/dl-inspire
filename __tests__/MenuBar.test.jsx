import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

import MenuBar from "../src/components/MenuBar";

test("MenuBar renders correctly", () => {
  const component = shallow(
    <MenuBar getUsers={() => {}} handleSearchTermChange={() => {}} />
  );
  expect(component).toMatchSnapshot();
});
