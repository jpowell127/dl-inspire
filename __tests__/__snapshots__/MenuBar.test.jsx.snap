// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuBar renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuBar
    getUsers={[Function]}
    handleSearchTermChange={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <MenuItem>
          <Header
            as="h1"
            color="pink"
          >
            <Icon
              as="i"
              name="behance"
            />
            <HeaderContent>
              Search
            </HeaderContent>
          </Header>
        </MenuItem>,
        <MenuMenu
          position="right"
        >
          <MenuItem>
            <Search
              getUsers={[Function]}
              handleSearchTermChange={[Function]}
            />
          </MenuItem>
        </MenuMenu>,
      ],
      "inverted": true,
      "size": "small",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Header
            as="h1"
            color="pink"
          >
            <Icon
              as="i"
              name="behance"
            />
            <HeaderContent>
              Search
            </HeaderContent>
          </Header>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "h1",
            "children": Array [
              <Icon
                as="i"
                name="behance"
              />,
              <HeaderContent>
                Search
              </HeaderContent>,
            ],
            "color": "pink",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "i",
                "name": "behance",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Search",
              },
              "ref": null,
              "rendered": "Search",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <MenuItem>
            <Search
              getUsers={[Function]}
              handleSearchTermChange={[Function]}
            />
          </MenuItem>,
          "position": "right",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Search
              getUsers={[Function]}
              handleSearchTermChange={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "getUsers": [Function],
              "handleSearchTermChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <MenuItem>
            <Header
              as="h1"
              color="pink"
            >
              <Icon
                as="i"
                name="behance"
              />
              <HeaderContent>
                Search
              </HeaderContent>
            </Header>
          </MenuItem>,
          <MenuMenu
            position="right"
          >
            <MenuItem>
              <Search
                getUsers={[Function]}
                handleSearchTermChange={[Function]}
              />
            </MenuItem>
          </MenuMenu>,
        ],
        "inverted": true,
        "size": "small",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Header
              as="h1"
              color="pink"
            >
              <Icon
                as="i"
                name="behance"
              />
              <HeaderContent>
                Search
              </HeaderContent>
            </Header>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "h1",
              "children": Array [
                <Icon
                  as="i"
                  name="behance"
                />,
                <HeaderContent>
                  Search
                </HeaderContent>,
              ],
              "color": "pink",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "i",
                  "name": "behance",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Search",
                },
                "ref": null,
                "rendered": "Search",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <MenuItem>
              <Search
                getUsers={[Function]}
                handleSearchTermChange={[Function]}
              />
            </MenuItem>,
            "position": "right",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Search
                getUsers={[Function]}
                handleSearchTermChange={[Function]}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "getUsers": [Function],
                "handleSearchTermChange": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
